mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -4777,10 +4777,33 @@ def _get_filter_clauses_for_search_traces(filter_string, session, dialect):
                 if key_name.startswith("attributes."):
                     attr_name = key_name[len("attributes.") :]
                     # Search within the content JSON for the specific attribute
-                    # Using JSON extraction with LIKE for broad database compatibility
-                    val_filter = SearchTraceUtils.get_sql_comparison_func(comparator, dialect)(
-                        SqlSpan.content, f'%"mlflow.spanAttribute.{attr_name}"{value}%'
-                    )
+                    # TODO: we should improve this by saving only the attributes into the table.
```

Comment:
Huge +1 to this - storing as a JSON column type would greatly simplify the brittle parsing logic for double-escaped serialized JSON that's in place now.